### ISSUE TITLE
Stitch rotated frames

### ISSUE THREAD

**Issue by Fatewarrior95** [spillerrec/Overmix#146]
I've been using Overmix for at least 3 years with very satisfying results, but I still can't find a proper way to stitch rotated pics into one, is this supported in any way?
And if not, is there at least a way to roughly get it done?
(Honestly, I haven't needed to use anything aside Aligning: Ordered and Cluster animation, so probably another option could help me but I don't know what do they do exactly)

I uploaded the frames, CAUTION: they are NSFW:
https://imgur.com/a/bCqofT0

**Comment by spillerrec** [comment]
No, neither rotation or scaling/zooming is supported currently. It is certainly something I want to add, but that have been the state for years. I will need to implement something which can detect the rotation/scale, I have been thinking of trying to do a Log-polar transform for this, but don't expect anything soon. There are a bunch of things I want to add, and I haven't made much progress on any of them.

Btw, have you tried the demosacing method for removing censorship? https://github.com/spillerrec/Overmix/wiki/Tutorial-%7C-Demosaicing

**Comment by Fatewarrior95** [comment]
Oh, I see, no problem.
And about the demosacing, I did try it a long time ago, when I first discovered the program, but I didn't get any result, I followed the steps closely but it didn't result in the end, I don't know why.
But I did find something very interesting about the way Overmix makes the stitches, I don't know if you knew about this.
With some types of censorships, when doing the stitches it kinda blurs it out and makes the censorship less mosaiced, What I mean is, Overmix kinda interpolates the mosaics and merges them, erasing very slightly the censorship. I think this could be really big if you could dig more into this.

Is better if I leave you some examples, again the pics are NSFW: (these are the best examples I got)
https://imgur.com/a/NYqOScv

**Comment by spillerrec** [comment]
Yes, these examples is exactly what the demosacing tries to improve upon. The first image would likely work very well, the second might give okayish results, while the last one doesn't do much because there is only vertical movement.

It was quite difficult to find something which worked in the past as it only worked if the mosaic sizes fit the pixel grid, but now you can use decimals to get the correct size and I made a small tool to help find the correct size and offset, making it much easier to get it working. The guide has been updated to match. If you use Overmix a lot for NSFW content I suggest you to try it again with the newest release, because while it can only be used in some situations, the results when it can are pretty amazing. Feel free to ask if you have difficulties.

Images often makes everything easier, but I'm not sure what GitHubs stance is to linking to explicit content. (You can use the e-mail in the ReadMe if that is an issue.) And it helps me spot other issues, for example the green corners in your first image. This should be fixed in #41, are you using the newest release? There is a workaround as well if it is still happening.

**Comment by FunkyJustin** [comment]
I was just about to post an issue like this but looks like somebody already has, I'm looking forward to being able to stitch frames that zoom out and rotate since I've been seeing a lot of them in animes lately.

**Comment by spillerrec** [comment]
Thanks for the feedback. I will try to see if I can actually get some progress on this done, but don't expect too much as I don't spend much time programming in my spare time currently.

If you are up to it, do you mind creating some samples containing rotation/zoom? I'm in the process of making a collection of short video clips containing scenes to stitch in order to automatically evaluate if aligning works correctly, which will make it easier for me to know if my changes improves or degrades the aligning step.
What you would have to do is to install FFMPEG, open the command line, and then run a command like this:
```bash
ffmpeg -i "<input file path>" -ss <hours>:<minutes>:<seconds> -t <duration> -vcodec copy "<output file.mkv>"
```
So to make a 10 seconds clip at 4 minutes and 20 seconds for the file `Kaguya-sama ep 07.mkv` you would write:
```bash
ffmepg -i "Kaguya-sama ep 07.mkv" -ss 0:04:20 -t 10 -vcodec copy "sample 01.mkv"
```
Due to limitations on where you can cut a video file the starting time might need to be a little earlier, just check the resulting file and if the beginning is missing then adjust the time a few seconds backwards and try again.

Example:
https://files.catbox.moe/1u3a8l.mkv
